src/legacy/core_plugins/vis_default_editor/public/components/controls/raw_json.tsx

```diff
@@ -47,34 +49,44 @@ function RawJsonParamEditor({
       />
     </>
   );
-  const isValid = isValidJson(value);
 
-  const onChange = (ev: React.ChangeEvent<HTMLTextAreaElement>) => {
-    const textValue = ev.target.value;
-    setValue(textValue);
-    setValidity(isValidJson(textValue));
+  const onChange = (newValue: string) => {
+    setValue(newValue);
   };
 
-  useEffect(() => {
-    setValidity(isValid);
-  }, [isValid]);
+  const onEditorValidate = (annotations: any[]) => {
+    // The first onValidate returned from EuiCodeEditor is a false positive
+    if (editorReady) {
+      const validity = annotations.length === 0;
+      setFormValidity(validity);
+      setValidity(validity);
+    } else {
+      setEditorReady(true);
+    }
+  };
 
   return (
     <EuiFormRow
       label={label}
-      isInvalid={showValidation ? !isValid : false}
+      isInvalid={showValidation ? !isFormValid : false}
       fullWidth={true}
       compressed
     >
-      <EuiTextArea
+      <EuiCodeEditor
         id={`visEditorRawJson${agg.id}`}
-        isInvalid={showValidation ? !isValid : false}
+        mode="json"
+        theme="github"
+        width="100%"
+        height="250px"
         value={value}
+        onValidate={onEditorValidate}
+        setOptions={{
+          fontSize: '14px',
+        }}
         onChange={onChange}
-        rows={2}
         fullWidth={true}
         onBlur={setTouched}
-        compressed
+        aria-label="Code Editor"
```

Comment:
I did some digging and hooking up `EuiCodeEditor` with `EuiFormRow` is a little difficult so we are going to have to duplicate the label similarly to how you have here but two suggestions on what you have hear:
- can you match the visual label? (In an ideal world, we wouldn't even have this `aria-label` and would use the proper label we use above so at least the text should be the same.)
- can you make sure this is translated?